repo/ref.go

```diff
@@ -344,7 +354,7 @@ func CanonicalizeDatasetRef(r Repo, ref *DatasetRef) error {
 
 // CanonicalizeProfile populates dataset DatasetRef ProfileID and Peername properties,
 // changing aliases to known names, and adding ProfileID from a peerstore
-func CanonicalizeProfile(r Repo, ref *DatasetRef) error {
+func CanonicalizeProfile(r Repo, ref *DatasetRef, need *NeedPeernameRenames) error {
```

Comment:
Adding `*NeedPeernameRenames` feels like overloading the purpose of `CanonicalizeProfile`, but I think we should merge this as-is.